geomstats/geometry/full_rank_matrices.py

```diff
@@ -22,12 +22,12 @@ class FullRankMatrices(OpenSet):
     def __init__(self, n, k, **kwargs):
         if "dim" not in kwargs.keys():
             kwargs["dim"] = n * k
-        super(FullRankMatrices, self).__init__(
-            ambient_space=Matrices(n, k), metric=MatricesMetric(n, k), **kwargs
-        )
+        super(FullRankMatrices, self).__init__(ambient_space=Matrices(n, k), **kwargs)
         self.rank = min(n, k)
         self.n = n
         self.k = k
+        if self._metric is None:
```

Comment:
I wonder if instead, we should do something like:

```
kwargs.setdefault('metric', MatricesMetric(n, k)
```

For dim we can do similarly:
```
kwargs.setdefault('dim', n*k)
```

Reply:
that's a good point, I didn' t know that syntax. There are places where we do 
```python
if dim not in kwargs:
    kwargs['dim'] = ...
```
And also for `shape` and `default_point_type`